packages/router-generator/src/generator.ts

```diff
@@ -365,6 +365,12 @@ export async function generator(config: Config) {
 
   const virtualRouteNodes = sortedRouteNodes.filter((d) => d.isVirtual)
 
+  const rootPathIdExtension = config.addExtensions && !config.disableTypes
+    ? '.tsx'
+    : config.addExtensions && config.disableTypes
+      ? '.jsx'
```

Comment:
why do we have to statically assign the file extension? why don't you just use the actual file extension of the file being imported? this would make this much more reliable

Reply:
You need to find out the actual file extension at generation time.
AFAIK the file could be named:
- __root.tsx
- __root.jsx
- __root.ts
- __root.js